frontend/src/pages/projects/screens/spawner/storage/usePreferredStorageClass.ts

```diff
@@ -0,0 +1,45 @@
+import * as React from 'react';
+import { AppContext } from '~/app/AppContext';
+import { StorageClassKind } from '~/k8sTypes';
+
+const usePreferredStorageClass = (): StorageClassKind | undefined => {
+  const {
+    dashboardConfig: {
+      spec: { notebookController },
+    },
+    storageClasses,
+  } = React.useContext(AppContext);
+
+  const defaultClusterStorageClasses = storageClasses.filter((storageclass) =>
+    storageclass.metadata.annotations?.['storageclass.kubernetes.io/is-default-class']?.includes(
+      'true',
+    ),
+  );
+
+  const configStorageClassName = notebookController?.storageClassName ?? '';
+
+  if (defaultClusterStorageClasses.length != 0) {
```

Comment:
```suggestion
  if (defaultClusterStorageClasses.length !== 0) {
```
We should have a linter for this, we have a ticket to do so... but this is something we should probably address. I think in this particular case, it'll never do funky things in JS... but `==`/`!=` will coerce the value into different types to satisfy the check.

Reply:
now I did not agree to many comments ... but this one here is partly true. I researched this, it does not have an impact, and indeed all other parts of the dashboard code use != and ==, but yeah, === and !=== I think is better.

"
The == operator performs a loose equality comparison that performs type coercion if necessary to make the comparison possible. The === operator, on the other hand, performs a strict equality comparison that does not perform type coercion and requires the operands to have the same type (as well as the same value)"

However, here, array.length type is an integer, no problem and in fact even good, don't you think (type coercion not really required)

But, yes, type coercion does not hut either here.

Reply:
JS has a lot of flexibility -- we try to ignore most of it, but you can definitely do funky things with a `.length` property. Doubtful we ever will, but this isn't the end of the world and I'm sure when we get the linter in place it will be auto-fixed.